File: app/src/main/java/com/waygo/data/base/contract/DatabaseContract.java
package com.waygo.data.base.contract;

import android.content.ContentValues;
import android.database.Cursor;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public interface DatabaseContract<T> {
    @NonNull String getCreateTable();
    @NonNull String getDropTable();
    @NonNull String getTableName();
    @Nullable T read(Cursor cursor);
    @NonNull ContentValues getContentValuesForItem(T item);
    @NonNull String[] getProjection();
}

File: app/src/main/java/com/waygo/data/base/contract/DatabaseContract.kt
package com.waygo.data.base.contract

import android.content.ContentValues
import android.database.Cursor

public interface DatabaseContract<T> {
    public fun getCreateTable(): String
    public fun getDropTable(): String
    public fun getTableName(): String
    public fun read(cursor: Cursor): T
    public fun getContentValuesForItem(item: T): ContentValues
    public fun getProjection(): Array<String>
}
